### PR TITLE
Server Sent Events

### PR DESCRIPTION
**Pull request by minocys**
### CHANGES
1. Add server sent event broadcast hub/controller
1. Remove image cropping
1. Broadcast successful image uploads
1. Respond with 200 status only on successful image uploads

### NOTES
Elected to use SSE's because: 
1. Cannot upload multiple files at once with websockets - multipart post ichiban bonzai
1. Application does not need bi direction communication, only broadcast
 
### CAVEATS
1. SSE not supported on IE and other legacy browser versions
1. Only 4-10 SSE connections can be opened per domain per browser

### CODE REVIEW

File: controllers/image.go [hunk 1]
```diff
@@ -22,46 +22,23 @@ type ImageController struct {
 	db *sql.DB
 	// Thumbnail Container Reference
 	tc *storage.Container
+	h  *Hub
 }
 
-// NewImageController returns an ImageController
-func NewImageController(db *sql.DB, tc *storage.Container) *ImageController {
+type NewImage struct {
```
Comment (hibearpanda) [inline controllers/image.go]: You can remove this struct type - it is not used at all

File: router/routes.go [hunk 2]
```diff
@@ -17,7 +17,14 @@ func InitRouter(db *sql.DB, tc *storage.Container, hc *storage.Container) *httpr
...
 	uc := controllers.NewUserController(db)
 	ic := controllers.NewInstanceController(db, hc)
-	imgC := controllers.NewImageController(db, tc)
+	hub := controllers.InitHub()
+	imgC := controllers.NewImageController(db, tc, hub)
+
+	// Start SSE hub
+	go hub.Listen()
+
+	//r.Handler("GET", "/register", hub)
```
Comment (hibearpanda) [inline router/routes.go]: Remove this comment

File: controllers/broadcast.go [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+	}()
+
+	notify := res.(http.CloseNotifier).CloseNotify()
+
+	for {
+		select {
+		case <-notify:
+			return
+		default:
+			fmt.Fprintf(res, "data: %s\n\n", <-c.conn)
```
Comment (hibearpanda) [inline controllers/broadcast.go]: Can remove this

Comment (minocys) [inline controllers/broadcast.go]: cannot, it's how data is sent

**Comment by minocys** [comment]
updated

**Pull request merged**